src/navigation/context.py

```diff
@@ -75,26 +75,38 @@ def current_fid_pos(self) -> Optional[np.ndarray]:
         """
         assert self.ctx.course
         current_waypoint = self.ctx.course.current_waypoint()
-        if current_waypoint is None or not self.ctx.course.look_for_post():
+        if current_waypoint is None:
             return None
 
         return self.get_fid_pos(current_waypoint.fiducial_id)
 
+    def other_gate_fid_pos(self) -> Optional[np.ndarray]:
+        """
+        Retrieves the position of the next fiducial (and we are looking for it) if we are looking for gate
+        """
+        assert self.ctx.course
+        current_waypoint = self.ctx.course.current_waypoint()
+        if current_waypoint is None or self.ctx.course.look_for_post():
```

Comment:
I think the logic would be more clear (and also would account for the case where this somehow gets called if we are just driving to a waypoint) if you just did 

```
if self.ctx.course.look_for_gate():
   return self.get.....
else:
   return None
```

Reply:
Yeah that's a fair point, prolly a bit more readable